src/webpub_manifest_parser/core/ast.py

```diff
@@ -0,0 +1,831 @@
+
+        :param height: Height of the linked resource in pixels
+        :type height: int
+
+        :param width: Width of the linked resource in pixels
+        :type width: int
+
+        :param duration: Duration of the linked resource in seconds
+        :type duration: float
+
+        :param bitrate: Bit rate of the linked resource in kilobits per second
+        :type bitrate: float
+
+        :param language: Expected language of the linked resource
+        :type language: List[str]
+
+        :param alternate: Alternate resources for the linked resource
+        :type alternate: List[Link]
+
+        :param children: Resources that are children of the linked resource, in the context of a given collection role
+        :type children: List[Link]
+        """
+        super(Link, self).__init__()
+
+        self.href = href
+        self.templated = templated
+        self.type = _type
+        self.title = title
+        self.rel = rel
+        self.properties = properties
+        self.height = height
+        self.width = width
+        self.duration = duration
+        self.bitrate = bitrate
+        self.language = language
+        self.alternate = alternate
+        self.children = children
+
+    def __hash__(self):
+        """Calculate the hash.
+
+        :return: Hash
+        :rtype: int
+        """
+        return hash(
+            (
+                self.href,
+                self.templated,
+                self.type,
+                self.title,
+                frozenset(self.rel),
+                self.properties,
+                self.height,
+                self.width,
+                self.duration,
+                self.bitrate,
+                frozenset(self.language)
+                if isinstance(self.language, list)
+                else self.language,
+                frozenset(self.alternate),
+                frozenset(self.children),
+            )
+        )
+
+    def __repr__(self):
+        """Return a string representation of the object.
+
+        :return: String representation
+        :rtype: str
+        """
+        return (
+            u"<Link("
+            u"href={0}, "
+            u"templated={1}, "
+            u"type={2}, "
+            u"title={3}, "
+            u"rel={4}, "
+            u"properties={5}, "
+            u"height={6}, "
+            u"width={7}, "
+            u"duration={8}, "
+            u"bitrate={9}, "
+            u"language={10}, "
+            u"alternate={11}, "
+            u"children={12}".format(
+                self.href,
+                self.templated,
+                self.type,
+                self.title,
+                self.rel,
+                self.properties,
+                self.height,
+                self.width,
+                self.duration,
+                self.bitrate,
+                self.language,
+                self.alternate,
+                self.children,
+            )
+        )
+
+
+class LinkList(Node, list):
```

Comment:
[In this document](https://github.com/readium/architecture/blob/master/proposals/004-publication-helpers-services.md#link-helpers) you will find a number of "helpers" functions that could be useful as well for `LinkList`. Note that this is a general guideline, but the naming should follow the convention of the target language. So `get_by_rel()` is fine for example.